Automattic/wp-calypso
# Newsletter categories: add subkey authentication for `useSubscribedNewsletterCategories` `useNewsletterCategorySubscriptionMutation`

## Proposed Changes

This PR adds subkey authentication for 2 endpoints that should be accessible from the Subscription Management portal.

Authentication in this portal is done by setting a subkey cookie. To be able to query API endpoints, this query needs to be passed in an Authorization header with `X-WPSUBKEY <subkey>` value.

There is already a similar method present in Reader, but it didn't feel best practice to import something from a totally unrelated part of Calypso. A follow-up project will be to refactor the queries that use it there to use this new TypeScript implementation.

## Testing Instructions

1. Apply this PR
2. Subscribe to a site with newsletter categories enabled, as a non-WPCOM user
3. Confirm the subscription
4. Do the subkey cookie dance:
- copy the value of the cookie, 
- Navigate to http://calypso.localhost:3000/subscriptions
- Paste `document.cookie = "subkey=VALUE"` into your console
- Refresh & behold your subscriptions
5. Click the site you've just subscribed to
6. You should see the categories listed:

![CleanShot 2023-09-21 at 16 26 18@2x](https://github.com/Automattic/wp-calypso/assets/528287/33833c7c-b660-4ac2-b64b-681c5276a8a1)

7. Change some categories, they should update

Now repeat for a WPCOM user, both in /subscriptions and Reader

---

By the way, you might see this in the console, this is not caused by this PR - but was already present:

![CleanShot 2023-09-21 at 16 22 14@2x](https://github.com/Automattic/wp-calypso/assets/528287/0d2b6aca-a491-4360-b0c5-32a1a4ff2bed)


## Pre-merge Checklist
